packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -1199,3 +1311,45 @@ class FollowerLayer extends ContainerLayer {
     properties.add(new TransformProperty('transform', getLastTransform(), defaultValue: null));
   }
 }
+
+/// A composited layer which annotates its children with a value.
+///
+/// These values can be retrieved using [Layer.find] with a given [Offset]. If
+/// a [Size] is provided to this layer, then find will check if the provided
+/// offset is within the bounds of the layer.
+class AnnotatedRegionLayer<T> extends ContainerLayer {
+  /// Creates a new annotated layer.
+  AnnotatedRegionLayer(this.value, {this.size});
+
+  /// The value returned by [find] if the offset is contained within this layer.
+  final T value;
+
+  /// The [size] is optionally used to clip the hit-testing of [find].
+  ///
+  /// If not provided, all offsets are considered to be contained within this
+  /// layer, unless a parent layer applies a clip.
+  final Size size;
+
+  @override
+  S find<S>(Offset regionOffset) {
+    final S result = super.find<S>(regionOffset);
+    if (result != null)
+      return result;
+    if (size != null && !size.contains(regionOffset))
+      return null;
+    if (T == S) {
+      final Object untypedResult = value;
+      final S typedResult = untypedResult;
+      return typedResult;
+    }
+    return super.find<S>(regionOffset);
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(new DiagnosticsProperty<T>('value', value));
+    if (size != null)
+      properties.add(new DiagnosticsProperty<Size>('size', size));
```

Comment:
you can always add this property, even if it's null. Just set `defaultValue` to null.

Reply:
Fixed